asf_floorplan_interpreter/pipeline/predict_floorplan.py

```diff
@@ -1,28 +1,201 @@
+                self.window_door_model = load_model_s3(self.window_door_model_name)
+            if any(
+                [
+                    label in self.labels_to_predict
+                    for label in [
+                        "KITCHEN",
+                        "LIVING",
+                        "RESTROOM",
+                        "BEDROOM",
+                        "GARAGE",
+                        "OTHER",
+                    ]
+                ]
+            ):
+                self.room_type_model = load_model_s3(self.room_type_model_name)
+
+    def predict_labels(self, image_url, correct_kitchen=True, conf_threshold=0):
+        """
+        Predict label segments and a counts of labels using the loaded models for a floorplan.
+
+        Arguments:
+            image_url (str): A URL or a local directory of your floorplan image.
+            correct_kitchen (bool): Whether to at a minimum predict 1 kitchen per floorplan (True) or not (False).
+            conf_threshold (float): The prediction confidence threshold for outputted labels.
+
+        Outputs:
+            labels (list): A list of the predicted image segments for each label, this is in the form
+                [{'label': 'DOOR', 'points': [[525.348, 319.445], ...], 'type': 'polygon', 'confidence': 0.9}, ...]
+            label_counts (dict): A summary of the counts of all labels found for this image. For example,
+                {'DOOR': 12, 'WINDOW': 10, 'LIVING': 2, 'KITCHEN': 2, 'BEDROOM': 3, 'RESTROOM': 1}
+        """
+        labels = []
+        if "ROOM" in self.labels_to_predict:
+            results = self.room_model(image_url, save=False, verbose=False)
+            labels += yolo_2_segments(results)
+        if "STAIRCASE" in self.labels_to_predict:
+            results = self.staircase_model(image_url, save=False, verbose=False)
+            labels += yolo_2_segments(results)
+        if any([label in self.labels_to_predict for label in ["DOOR", "WINDOW"]]):
+            results = self.window_door_model(image_url, save=False, verbose=False)
+            labels += yolo_2_segments(results)
+        if any(
+            [
+                label in self.labels_to_predict
+                for label in [
+                    "KITCHEN",
+                    "LIVING",
+                    "RESTROOM",
+                    "BEDROOM",
+                    "GARAGE",
+                    "OTHER",
+                ]
+            ]
+        ):
+            results = self.room_type_model(image_url, save=False, verbose=False)
+            labels += yolo_2_segments(results)
+
+        # Remove any labels you aren't asking for and only if the prediction probability is over a threshold
+        labels = [
+            label
+            for label in labels
+            if (
+                (label["label"] in self.labels_to_predict)
+                and (label["confidence"] >= conf_threshold)
```

Comment:
```suggestion
        label_counts = Counter(
            segment["label"]
            for segment in filtered_segments
        )
```

Reply:
I'm sorry - I was going through the comments Clare suggested and I don't think I refreshed the page or something, so have just seen this (via checking my emails!)